core/src/main/java/org/elasticsearch/index/shard/IndexingOperationListener.java

```diff
@@ -40,7 +40,7 @@
     /**
      * Called after the indexing operation occurred.
      */
-    default void postIndex(Engine.Index index, boolean created) {}
+    default void postIndex(Engine.Index index, Engine.IndexResult result) {}
 
     /**
      * Called after the indexing operation occurred with exception.
```

Comment:
can we document the difference with the other postIndex and how failures are treated?